database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -97,6 +97,7 @@ public void onChildMoved(DataSnapshot snapshot, String previousChildKey) {
         mSnapshots.remove(oldIndex);
         int newIndex = previousChildKey == null ? 0 : (getIndexForKey(previousChildKey) + 1);
         mSnapshots.add(newIndex, snapshot);
+        mSnapshotMap.put(snapshot.getKey(), newIndex);
```

Comment:
Wouldn't `onChildMoved` affect the indexes of every key?  I have a feeling that's why @puf didn't use a `Map` in the first place but I could be wrong.